test/gtest/ucp/test_ucp_worker.cc

```diff
@@ -928,6 +930,52 @@ UCS_TEST_P(test_pci_bw, get_pci_bw)
 
 UCP_INSTANTIATE_TEST_CASE_TLS(test_pci_bw, all, "all")
 
+class rlimit_setter {
+public:
+    rlimit_setter() : m_resource(0), m_limit(0), m_orig({})
+    {
+    }
+
+    void set(int resource, rlim_t new_limit)
+    {
+        struct rlimit limit;
+
+        m_resource = resource;
+        if (getrlimit(m_resource, &m_orig) != 0) {
+            UCS_TEST_MESSAGE << "could not get rlimit";
+            return;
+        }
+
+        limit.rlim_cur = new_limit > m_orig.rlim_max ? m_orig.rlim_max :
+                                                       new_limit;
+        limit.rlim_max = m_orig.rlim_max;
+
+        if (setrlimit(m_resource, &limit) == 0) {
+            m_limit = limit.rlim_cur;
+            UCS_TEST_MESSAGE << "set rlimit to " << m_limit;
+        } else {
+            UCS_TEST_MESSAGE << "could not set rlimit to " << limit.rlim_cur;
+        }
+    }
+
+    ~rlimit_setter()
+    {
+        if (m_limit != 0) {
```

Comment:
with RAII with unique_ptr this check is redundant

Reply:
added unique_ptr, but we still need to check if we were able to getrlimit() in the first place